app/assets/javascripts/admin/addon/components/schema-theme-setting/types/categories.gjs

```diff
@@ -0,0 +1,73 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { and, not } from "truth-helpers";
+import I18n from "discourse-i18n";
+import FieldInputDescription from "admin/components/schema-theme-setting/field-input-description";
+import CategorySelector from "select-kit/components/category-selector";
+
+export default class SchemaThemeSettingTypeCategories extends Component {
+  @tracked touched = false;
+
+  @tracked
+  value =
+    this.args.value?.map((categoryId) => {
+      return this.args.setting.metadata.categories[categoryId];
+    }) || [];
+
+  required = this.args.spec.required;
+  min = this.args.spec.validations?.min;
+  max = this.args.spec.validations?.max;
+
+  @action
+  onInput(categories) {
+    this.touched = true;
+    this.value = categories;
+    this.args.onChange(categories.map((category) => category.id));
+  }
+
+  get validationErrorMessage() {
+    if (!this.touched) {
+      return;
+    }
+
+    if (this.min && this.value.length < this.min) {
+      return I18n.t("admin.customize.theme.schema.fields.categories.at_least", {
+        count: this.min,
+      });
+    }
+
+    if (this.required && (!this.value || this.value.length === 0)) {
+      return I18n.t("admin.customize.theme.schema.fields.categories.at_least", {
+        count: 1,
+      });
+    }
+  }
+
+  get categorySelectorOptions() {
```

Comment:
I think this can be inlined in the template. I think that makes it a bit easier to read and there is not really any complex logic to warrant a different method/getter.